src/id3manager/utils.py

```diff
@@ -23,3 +31,72 @@ def ms_to_human_time(ms: int) -> str:
 
 def sec_to_ms(sec: float):
     return int(sec * 1000)
+
+
+def parse_apic(
+    value: str,
+    mime: t.Optional[str] = None,
+    type: id3.PictureType = id3.PictureType.COVER_FRONT,
+) -> id3.APIC:
+    result = urlparse.urlparse(value)
+    match result.scheme:
+        case "data":
+            # Data URL (https://developer.mozilla.org/en-US/docs/Web/URI/Schemes/data).
+            # Image data is embedded in the URL as a base64 value and stored alongside
+            # the MIME type.
+            #
+            # E.g. "data:image/png;base64,SGVsbG8sIFdvcmxkIQ==".
+            try:
+                mime, encoded_value = result.path.split(";", maxsplit=1)
+                encoding, data = encoded_value.split(",", maxsplit=1)
+            except ValueError:
+                raise ValueError(f"Invalid data URL value: `{value}`")
+
+            if encoding != "base64":
+                raise ValueError(f"Unsupported encoding: `{encoding}`")
+
+            return id3.APIC(mime=mime, data=base64.b64decode(data), type=type)
+        case "file":
+            # URL specifying the location of an image file on the local host. The data
+            # is loaded to be stored in the APIC frame. The MIME type is deducted from
+            # the file extension (if not provided explicitly).
+            #
+            # E.g. "file:///path/to/image.jpeg"
+            if not mime:
+                mime, _ = mimetypes.guess_type(value)
+
+            return id3.APIC(
+                mime=mime,
+                data=open(result.path, "rb").read(),
+                type=id3.PictureType.COVER_FRONT,
```

Comment:
Seems like a bug. I believe `type=type` is expected here, isn't it?

Reply:
Good catch!